tests/trainer/test_supporters.py

```diff
@@ -325,22 +332,46 @@ def __len__(self):
         def __getitem__(self, index):
             return self.data[index]
 
+    class CustomSampler(RandomSampler):
+        def __init__(self, data_source, name) -> None:
+            super().__init__(data_source)
+            self.name = name
+
+    dataset = CustomDataset(range(10))
     dataloader = CombinedLoader(
         {
             "a": DataLoader(CustomDataset(range(10))),
+            "a": DataLoader(dataset, sampler=CustomSampler(dataset, "custom_sampler")),
             "b": {"c": DataLoader(CustomDataset(range(10))), "d": DataLoader(CustomDataset(range(10)))},
             "e": [DataLoader(CustomDataset(range(10))), DataLoader(CustomDataset(range(10)))],
         }
     )
 
-    trainer = Trainer(replace_sampler_ddp=True, accelerator="ddp", gpus=2)
-    dataloader = trainer.auto_add_sampler(dataloader, shuffle=True)
-    _count = 0
+    with mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": str(int(use_fault_tolerant))}):
+
+        trainer = Trainer(replace_sampler_ddp=replace_sampler_ddp, accelerator="ddp", gpus=2)
+        dataloader = trainer.prepare_dataloader(dataloader, shuffle=True)
+        _count = 0
 
     def _assert_distributed_sampler(v):
         nonlocal _count
         _count += 1
-        assert isinstance(v, DistributedSampler)
+        if use_fault_tolerant:
+            assert isinstance(v, FastForwardSampler)
+            v = v._sampler
```

Comment:
does this do anything?

Reply:
Yes, it extracts the wrapped sampler to make sure it is a distributed sampler. Added a check to make sure we pass within the part of the code.